probe-rs/src/probe.rs

```diff
@@ -374,7 +375,7 @@ impl Probe {
         &mut self,
         scan_chain: Vec<ScanChainElement>,
     ) -> Result<(), DebugProbeError> {
-        if self.attached {
+        if !self.attached {
```

Comment:
This might not be right, I had a hard time reasoning when the scan chain should be set.

Reply:
At least this blames to you :)

Reply:
Resolved with the latest commit